### PR TITLE
feat: add copy_to_bin rule

### PR DESCRIPTION
**Pull request by gregmagolan**
Will be useful in rules_js

### CODE REVIEW

File: docs/copy_to_bin.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<!-- Generated with Stardoc: http://skydoc.bazel.build -->
+
+AA rule that copies source files to the output tree.
```
Comment (alexeagle) [inline docs/copy_to_bin.md:3]: double A

File: docs/copy_to_bin.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+This rule uses a Bash command (diff) on Linux/macOS/non-Windows, and a cmd.exe
```
Comment (alexeagle) [inline docs/copy_to_bin.md:5]: maybe a note about where we copied it from?

Comment (alexeagle) [reply docs/copy_to_bin.md:5]: there was a thread on bazel-skylib about it with Yun

**Review by alexeagle — APPROVED** [review]

**Review by kormide — APPROVED** [review]

File: lib/copy_to_bin.bzl [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"""AA rule that copies source files to the output tree.
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply lib/copy_to_bin.bzl:1]: Whoops.

File: docs/copy_to_bin.md (hunk 2 above)
Comment (gregmagolan) [reply docs/copy_to_bin.md:5]: Will note that we pulled it from rules_nodejs

**Pull request merged**